app/main.py

```diff
@@ -1,25 +1,72 @@
 from abc import ABC
+from typing import Type
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: object, owner: object) -> str:
```

Comment:
The return type annotation for the __get__ method should be `int`, not `str`. Change the return type to `int`.